src/commands/revisionDraft/RevisionDraftDeployPromptStep.ts

```diff
@@ -0,0 +1,48 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { AzureWizardPromptStep, type IAzureQuickPickItem } from "@microsoft/vscode-azext-utils";
+import { showDraftCommandDeployPopupSetting } from "../../constants";
+import { localize } from "../../utils/localize";
+import { settingUtils } from "../../utils/settingUtils";
+import { type RevisionDraftContext } from "./RevisionDraftContext";
+
+export class RevisionDraftDeployPromptStep<T extends RevisionDraftContext> extends AzureWizardPromptStep<T> {
+    public async prompt(context: T): Promise<void> {
+        if (!await settingUtils.getGlobalSetting(showDraftCommandDeployPopupSetting)) {
+            context.shouldDeployRevisionDraft = false;
+            return;
+        }
```

Comment:
Should you try to move this into `shouldPrompt`? To do so, you could set this context before running the wizard.

Reply:
Yeah good call